netket/graph/hypergrid.py

```diff
@@ -0,0 +1,132 @@
+from graph import NetworkX
+
+import numpy as _np
+import networkx as _nx
+
+
+class Hypergrid(NetworkX):
+    r"""A hypergrid lattice of d dimensions, and possibly different sizes of each dimension. 
+    Periodic boundary conditions can also be imposed"""
+
+    def __init__(self, length, pbc=True):
+        """
+        Constructs a new ``Grid`` given its length vector.
+
+        Args:
+            length: Side length of the hypergrid. It must be a list with integer components >= 1.
+            pbc: If ``True```then the constructed hypergrid will have
+                periodic boundary conditions, otherwise open boundary conditions
+                are imposed.
+        Examples:
+            A 5x10 lattice with periodic boundary conditions can be
+            constructed as follows:
+
+            >>> import netket
+            >>> g=netket.graph.Hypergrid(length=[5, 10], pbc=True)
+            >>> print(g.n_nodes)
+            50
+
+            Also, a 2x2x3 lattice with open boundary conditions can be constructed as follows:
+            >>> g=netket.graph.Hypergrid(length=[2,2,3], pbc=False)
+            >>> print(g.n_nodes)
+            12
+        """
+
+        try:
+            assert isinstance(length, list)
+        except AssertionError:
+            raise TypeError("length must be a list of integers >= 1")
+
+        try:
+            condition = [isinstance(x, int) and x >= 1 for x in length]
+            if sum(condition) != len(length):
+                raise TypeError
+        except TypeError:
+            raise TypeError("Components of length must be integers greater than 1")
```

Comment:
Same as above.
This doesn't even change the type of the error. Just raise the exception below the if, not need to catch and re-raise. Also, the contents of the lists being wrong should bey a `ValueError`.

Reply:
Sorry, forgot to change those.